staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go

```diff
@@ -17,20 +17,49 @@ limitations under the License.
 package customresource
 
 import (
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
 	"k8s.io/apiserver/pkg/registry/generic"
 	genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
+	"k8s.io/apiserver/pkg/registry/rest"
+	scalescheme "k8s.io/client-go/scale/scheme"
```

Comment:
This was intended to be *purely* internal to the scale client.  The internal types aren't supposed to be used by other people.

Reply:
Make them `internal`.

Reply:
you mean change the name?  Because they are marked as having the version be `__internal`.

Reply:
We don't make any use of that, but Go 1.5+ has packages called `internal` which can only be imported if your package and the internal one shares a common root. Does not belong into this PR, but I really think we should move a lot of internal code into internal packages. These client schemes are candidates for that. So I agree, we probably should not use the scale scheme here in the registry.

Reply:
Uses external type for autoscaling now. :+1: